### PR TITLE
Bug fixes for targeting a world with resource borrows

### PR DESCRIPTION
**Pull request by calvinrp**
Expanded test cases to cover issue #299 

Also, when specifying an exact semver requirement, adds that exact requirement to the generated `Cargo.toml`. There was mismatch before.

`wit-bindgen-rust` generates different bindings for `borrow<>` of resources, depending on whether that type is exported in the world or not. `&SomeType` vs `SomeTypeBorrow` in the generated bindings. Fixed the logic here to match that behavior. The revised test case should better cover future potential breakage.

If interested, here is some of the logic on the `wit-bindgen` side:
https://github.com/bytecodealliance/wit-bindgen/blob/main/crates/rust/src/interface.rs#L2203-L2225

### CODE REVIEW

**Review by macovedj — APPROVED** [review]

File: src/generator.rs [hunk 1]
```diff
@@ -416,28 +405,54 @@ impl<'a> UnimplementedFunction<'a> {
...
             }
-            TypeDefKind::Type(ty) => self.print_type(in_world, ty, trie, source)?,
-            TypeDefKind::Handle(Handle::Own(id)) => {
-                self.print_type_id(in_world, *id, trie, source)?
-            }
+            TypeDefKind::Type(ty) => self.print_type(ty, trie, source)?,
+            TypeDefKind::Handle(Handle::Own(id)) => self.print_type_id(*id, trie, source, false)?,
             TypeDefKind::Handle(Handle::Borrow(id)) => {
-                if in_world {
+                let type_info = &self.resolve.types[*id];
```
Comment (fibonacci1729) [inline src/generator.rs:421]: nit: Could we pull the body of this case out into a method called `is_export_resource` (or something)? Keeps it a little more readable IMO.

**Review by fibonacci1729 — APPROVED** [review]
Minor nit, but overall makes sense to me!

File: src/generator.rs (hunk 1 above)
Comment (calvinrp) [reply src/generator.rs:421]: Yep, good call

Comment (calvinrp) [reply src/generator.rs]: Fixed, thanks

**Pull request merged**